src/config.js

```diff
@@ -0,0 +1,51 @@
+const Joi = require("@hapi/joi");
+
+const fragmentsTypesSchema = Joi.object({
+  title: Joi.string().required(),
+  extension: Joi.string().required()
+}).required();
+
+const schema = Joi.object({
+  changelogFile: Joi.string().required(),
+  changelogDateFormat: Joi.string().required(),
+  changelogTemplate: Joi.string().required(),
+  fragmentsFolder: Joi.string().required(),
+  fragmentsTypes: Joi.array()
+    .items(fragmentsTypesSchema)
+    .required()
+});
+
+const changelogTemplate = `# [newVersion] - (bumpDate)
+{{#each fragments}}
+## {fragment.title}
+  {{#each fragmentEntries}}
+    * {fragment}
+  {{/each}}
+{{/each}}
+`;
+
+const baseConfig = {
+  changelogFile: "CHANGELOG.md",
+  changelogDateFormat: "YYYY-MM-DD",
+  changelogTemplate: changelogTemplate,
+  fragmentsFolder: "fragments",
+  fragmentsTypes: [
+    { title: "Features", extension: "feature" },
+    { title: "Bugfixes", extension: "bugfix" },
+    { title: "Documentation", extension: "doc" },
+    { title: "Deprecations and Removals", extension: "removal" },
+    { title: "Misc", extension: "misc" }
+  ]
+};
+
+module.exports.buildConfig = function(config) {
+  newsFragmentConfiguration = Object.assign({}, baseConfig, config);
+
+  const { error } = schema.validate(newsFragmentConfiguration);
+
+  if (error) {
+    throw new Error(error.message);
```

Comment:
Talvez seja legal começarmos a colocar uns erros mais específicos, tipo um `ValidationError`. O que acha?

Reply:
@matheusfiorin maybe. We'll use this `ValidationError` in some other place? Maybe we could keep it simple for now.

Reply:
I don't think that we're going to use in a lot of places, makes sense to leave it for now.